src/main/scala/diplomacy/Nodes.scala

```diff
@@ -658,11 +762,39 @@ class SinkNode[D, U, EO, EI, B <: Data](imp: NodeImp[D, U, EO, EI, B])(pi: Seq[U
 {
   override def description = "sink"
   protected[diplomacy] def resolveStar(iKnown: Int, oKnown: Int, iStars: Int, oStars: Int): (Int, Int) = {
-    require (iStars <= 1, s"$context appears left of a :*= ${iStars} times; at most once is allowed")
-    require (oStars == 0, s"$context cannot appear right of a :=*")
-    require (oKnown == 0, s"$context cannot appear right of a :=")
-    require (pi.size == iKnown || iStars == 1, s"$context has only ${iKnown} inputs connected out of ${pi.size}")
-    require (pi.size >= iKnown, s"$context has ${iKnown} inputs out of ${pi.size}; cannot assign ${pi.size - iKnown} edges to resolve :*=")
+    def resolveStarInfo: String =
+      s"""$context
+         |$bindingInfo
+         |inward nodes already known: $iKnown
+         |outward nodes already known: $oKnown
+         |query by nodes other nodes: $iStars
+         |query to other nodes: $oStars
+         |${pi.size} upward parameters: [${pi.map(_.toString).mkString(",")}]
+         |""".stripMargin
+    require (iStars <= 1,
+      s"""$resolveStarInfo
+         |This node appears left of a :*= $iStars times; at most once is allowed.
+         |""".stripMargin)
+    require (oStars == 0,
+      s"""$resolveStarInfo
+         |This node cannot appear right of a :*=
+         |""".stripMargin)
+    require (oKnown == 0,
+      s"""$resolveStarInfo
+         |This node cannot appear right of a :=
+         |""".stripMargin)
+    if (iStars == 0)
+      require(pi.size == iKnown,
+        s"""$resolveStarInfo
+           |$oKnown inputs were connected but ${pi.size} sinks were specified to the node constructor.
```

Comment:
`|The following node had $oKnown inward bindings connected to it, but ${pi.size} sinks were specified to the node constructor.`

Reply:
Note: iKnown not oKnown